src/system/zephyr/system.c

```diff
@@ -103,7 +103,13 @@ int8_t _z_condvar_wait(_z_condvar_t *cv, _z_mutex_t *m) { return pthread_cond_wa
 /*------------------ Sleep ------------------*/
 int z_sleep_us(unsigned int time) { return usleep(time); }
 
-int z_sleep_ms(unsigned int time) { return usleep(time * 1000U); }
+int z_sleep_ms(unsigned int time) {
+    for (unsigned int i = 0; i < time; i++) {
+        z_sleep_us(1000);
+    }
+
+    return 0;
+}
```

Comment:
As discussed in #156, I had a [fix](https://github.com/dcorbeil/zenoh-pico/commit/b67b34013c531c88fcbc18ffe9f0a4dc80780570) for this but you were faster on the trigger than me. Feel free to use it if you wish.

My thinking was to keep using Zephyr's POSIX API as you were already doing for `z_sleep_us` and `z_sleep_s` and because the now deprecated [`usleep`](https://man7.org/linux/man-pages/man3/usleep.3.html) recommends using `nanosleep` instead, I went with that.

Although, under the hood they end up doing mostly the same thing.

Reply:
You can still do the PR. We really appreciate the engagement from the community and users.

I believe you can do the PR on top of this one, and you would appear as co-author when merged. I also departed from your feedback to fix an eventual issue on the other platforms.

Reply:
Ok I'll try that.

Reply:
Thanks. I have accepted it into this PR.
However, in order to be merged upstream into the Eclipse repositories you need to sign the ECA with the same email used in the commit message. Check more details here: https://accounts.eclipse.org/legal/eca/validation/138875